x/emissions/keeper/params.go

```diff
@@ -5,14 +5,25 @@ import (
 	"github.com/zeta-chain/zetacore/x/emissions/types"
 )
 
-// GetParamsIfExists get all parameters as types.Params if they exist
-// non existent parameters will return zero values
-func (k Keeper) GetParamsIfExists(ctx sdk.Context) (params types.Params) {
-	k.paramStore.GetParamSetIfExists(ctx, &params)
-	return
+// GetParams get all parameters
+func (k Keeper) GetParams(ctx sdk.Context) (params types.Params, found bool) {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.KeyPrefix(types.ParamsKey))
+	if bz == nil {
+		return types.Params{}, false
+	}
+	k.cdc.MustUnmarshal(bz, &params)
+	return params, true
 }
 
 // SetParams set the params
-func (k Keeper) SetParams(ctx sdk.Context, params types.Params) {
-	k.paramStore.SetParamSet(ctx, &params)
+func (k Keeper) SetParams(ctx sdk.Context, params types.Params) error {
+	if err := params.Validate(); err != nil {
+		return err
+	}
+
+	store := ctx.KVStore(k.storeKey)
+	bz := k.cdc.MustMarshal(&params)
```

Comment:
Can we use `Marshal` and return the error here?

Reply:
I think for set params, we are returning err already, 
for get, can we try returning false on error, I guess handling the error is slightly safer than panicking, wdyt?